src/languages/ruby.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  ancestorChainNodeMatcher,
+  argumentMatcher,
+  cascadingMatcher,
+  conditionMatcher,
+  createPatternMatchers,
+  leadingMatcher,
+  patternMatcher,
+  trailingMatcher,
+} from "../util/nodeMatchers";
+import {
+  NodeMatcherAlternative,
+  ScopeType,
+  SelectionWithEditor,
+} from "../typings/Types";
+import { SyntaxNode } from "web-tree-sitter";
+import { getNodeRange } from "../util/nodeSelectors";
+import { patternFinder } from "../util/nodeFinders";
+
+// Generated by the following command:
+// > curl https://raw.githubusercontent.com/tree-sitter/tree-sitter-ruby/1ebfdb288842dae5a9233e2509a135949023dd82/src/node-types.json \
+// | jq '[.[] | select(.type == "_statement" or .type == "_simple_statement") | .subtypes[].type]'
+const STATEMENT_TYPES = [
+  "_expression",
+  "alias",
+  "begin_block",
+  "end_block",
+  "if_modifier",
+  "rescue_modifier",
+  "undef",
+  "unless_modifier",
+  "until_modifier",
+  "while_modifier",
+];
+
+// Generated by the following command:
+// > curl https://raw.githubusercontent.com/tree-sitter/tree-sitter-ruby/1ebfdb288842dae5a9233e2509a135949023dd82/src/node-types.json \
+//   | jq '[.[] | select(.type == _expression or .type == _arg or .type == _primary or .type == _lhs or .type == _simple_numeric or .type == _variable or .type == _nonlocal_variable) | .subtypes[].type | select(startswith(_) | not)] | sort | unique'
+const EXPRESSION_TYPES = [
+  "array",
+  "assignment",
+  "begin",
+  "binary",
+  "break",
+  "call",
+  "case",
+  "case_match",
+  "chained_string",
+  "character",
+  "class",
+  "class_variable",
+  "complex",
+  "conditional",
+  "constant",
+  "delimited_symbol",
+  "element_reference",
+  "false",
+  "float",
+  "for",
+  "global_variable",
+  "hash",
+  "heredoc_beginning",
+  "identifier",
+  "if",
+  "instance_variable",
+  "integer",
+  "lambda",
+  "method",
+  "module",
+  "next",
+  "nil",
+  "operator_assignment",
+  "parenthesized_statements",
+  "range",
+  "rational",
+  "redo",
+  "regex",
+  "retry",
+  "return",
+  "scope_resolution",
+  "self",
+  "simple_symbol",
+  "singleton_class",
+  "singleton_method",
+  "string",
+  "string_array",
+  "subshell",
+  "super",
+  "symbol_array",
+  "true",
+  "unary",
+  "unless",
+  "until",
+  "while",
+  "yield",
+];
+
+// TODO: Add more allowable parent types
+const EXPRESSION_STATEMENT_PARENT_TYPES = ["program", "method"];
+
+const mapTypes = ["hash"];
+const listTypes = ["array", "string_array", "symbol_array"];
+
+const assignmentOperators = ["=", '+=', '-=', '*=', '**=', '/=', '||=', '|=', '&&=', '&=', '%=', '>>=', '<<=', '^='];
+const mapKeyValueSeparators = [":", "=>"];
```

Comment:
@pokey — I opted to go with terminating the leading `:` as well since Ruby is odd in this case and it would be better to be inline with how other languages behave.